sklearn/utils/estimator_checks.py

```diff
@@ -368,55 +378,179 @@ def check_transformer_pickle(name, Transformer):
+        assert_raises(ValueError, estimator.partial_fit, X[:, :-1], y)
+    elif isinstance(estimator, TransformerMixin):
+        estimator.partial_fit(X)
+        assert_raises(ValueError, estimator.partial_fit, X[:, :-1])
+
+
+def check_estimators_pfit_return(name, Estimator):
+    # Check if estimators with partial fit capability return reference to 
+    # self at the end of partial_fit call.
+    if not hasattr(Estimator, 'partial_fit'):
         return
     X, y = make_blobs(n_samples=50, random_state=1)
     X -= X.min()
     with warnings.catch_warnings(record=True):
-        alg = Alg()
-    set_fast_parameters(alg)
-    if isinstance(alg, ClassifierMixin):
+        estimator = Estimator()
+    set_fast_parameters(estimator)
+    if isinstance(estimator, ClassifierMixin):
         classes = np.unique(y)
-        alg.partial_fit(X, y, classes=classes)
+        ret_val = estimator.partial_fit(X, y, classes=classes)
+    elif isinstance(estimator, RegressorMixin):
+        ret_val = estimator.partial_fit(X, y)
     else:
-        alg.partial_fit(X, y)
+        ret_val = estimator.partial_fit(X)
+
+    assert_equal(ret_val, estimator)
+
+
+def check_estimators_pfit_reset_when_fit(name, Estimator):
+    # Check if the estimators get reset when fit is applied after a
+    # partial fit.
+    # Estimators are assumed to be reset when the transorm / predic result
+    # is similar to 1st fit call
+    skip_list = ('MiniBatchKMeans')
+    if not hasattr(Estimator, 'partial_fit') or name in skip_list:
+        return
+
+    X, y = make_blobs(n_samples=50, random_state=1)
+    X -= X.min()
+
+    with warnings.catch_warnings(record=True):
+        estimator = Estimator()
+
+    set_fast_parameters(estimator)
+    classes = np.unique(y)
+
+    # Initial call to fit
+    try:
+        estimator.fit(X)
+    except TypeError:
+        estimator.fit(X, y)
+
+    if hasattr(estimator, 'transform'):
+        result_first_fit = estimator.transform(X)
+    elif hasattr(estimator, 'predict'):
+        result_first_fit = estimator.predict(X)
+
+    # Run 2 iterations of full set of partial_fit with batches of size 10
+    for i in range(2):
+        X_s, y_s = shuffle(X, y, random_state=0)
+        for j in range(0, 50, 10):
+            if isinstance(estimator, ClassifierMixin):
+                estimator.partial_fit(X_s[j:10+j], y_s[j:10+j], classes=classes)
+            elif isinstance(estimator, RegressorMixin):
+                estimator.partial_fit(X_s[j:10+j], y_s[j:10+j])
+            else:
+                estimator.partial_fit(X_s[j:10+j])
+
+    # Finally call fit to reset the estimator
+    try:
+        estimator.fit(X)
+    except TypeError:
+        estimator.fit(X, y)
+
+    if hasattr(estimator, 'transform'):
+        result_second_fit = estimator.transform(X)
+    elif hasattr(estimator, 'predict'):
+        result_second_fit = estimator.predict(X)
+
+    assert_array_equal(result_second_fit, result_first_fit)
```

Comment:
Actually, I've changed my mind. I made that statement only after a skim. Checking that changing the number of features for `fit` works is sufficient, methinks.